arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -78,9 +68,46 @@ def run_timerlat(
                 f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
             )
 
+        if params.enable_time_series:
+            timeseries_dict = {}
+            trace_path = "/sys/kernel/debug/tracing/instances/timerlat_hist/trace_pipe"
+
+            timeseries_cmd = ["cat", trace_path]
+
+            # A delay is needed before reading from the trace_path to ensure the file
+            # exists and data is streaming to the pipe. I've tried to avoid a sleep(),
+            # but none of the methods I've tested have worked.
+            wait_time = 0
+            timeout_seconds = 5
+            sleep_time = 0.5
+            trace_file_exists = False
+            while not trace_file_exists and wait_time < timeout_seconds:
+                time.sleep(sleep_time)
+                trace_file_exists = os.path.isfile(trace_path)
+                wait_time += sleep_time
+
+            if not trace_file_exists:
```

Comment:
I think I would use an absolute time deadline instead of a counted number of intervals; and, I would avoid capturing the `isfile()` result in a local variable, since only the last call really matters.  This reduces the loop setup and makes the loop body much simpler.

```suggestion
            timeout = datetime.now() + timedelta(seconds=5)
            while not os.path.isfile(trace_path) and datetime.now() < timeout:
                time.sleep(0.5)

            if not os.path.isfile(trace_path):
```